platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/service/schemas/V057PlatformStateSchema.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.service.schemas;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.swirlds.state.lifecycle.MigrationContext;
+import com.swirlds.state.lifecycle.Schema;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.function.Supplier;
+
+/**
+ * FUTURE: A restart-only schema that ensures the platform state at startup
+ * reflects the active roster.
+ * <p>
+ * C.f. <a href="https://github.com/hashgraph/hedera-services/issues/16552">here</a>.
+ */
+public class V057PlatformStateSchema extends Schema {
+    private static final SemanticVersion VERSION =
+            SemanticVersion.newBuilder().major(0).minor(57).build();
+
+    /**
+     * A supplier for the active roster.
+     */
+    private final Supplier<Roster> activeRoster;
+
+    public V057PlatformStateSchema(@NonNull final Supplier<Roster> activeRoster) {
+        super(VERSION);
+        this.activeRoster = requireNonNull(activeRoster);
```

Comment:
We're currently working on removing the AddressBook objects (both current and previous) from the PlatformState, and currently we don't have plans to reintroduce similar concepts for rosters in the PlatformState. The RosterService should be the only source of truth for the information about the active/previous rosters. So this supplier doesn't seem to belong here.

Any code that needs to know the roster(s), should either use the RosterService directly, or use the `RosterRetriever` depending on the context (ideally, we'd deprecate the RosterRetriever eventually and only ever use the RosterService, but currently there's special cases where we need to use the retriever for now - e.g. in https://github.com/hashgraph/hedera-services/pull/16459 ).

Reply:
True, once this platform migration is done, we may want a different migration here.

But since we hope to merge this PR and then almost immediately enable the `Roster` lifecycle in PR checks, we need a stopgap here.

Reply:
Sounds good.